app/(tabs)/_layout.tsx

```diff
@@ -1,14 +1,41 @@
-import { Tabs } from 'expo-router';
+import { type Href, Link, Tabs } from 'expo-router';
+import { StyleSheet, TouchableOpacity } from 'react-native';
 import { TabBarIcon } from '../../components/TabBarIcon';
 import { colors } from '../../constants/colors';
 
+const styles = StyleSheet.create({
+  headerRight: {
+    marginRight: 16,
+  },
+});
+
+type Props = {
+  href: Href<string>;
+};
+
+function HeaderRight({ href }: Props) {
+  return (
+    <Link href={href} asChild>
+      <TouchableOpacity style={styles.headerRight}>
+        <TabBarIcon name="add" color={colors.text} />
+      </TouchableOpacity>
+    </Link>
+  );
+}
+
 export default function TabLayout() {
+  const renderHeaderRight = (href: Href<string>) => <HeaderRight href={href} />;
```

Comment:
This seems like a weird pattern - let's just inline this with hardcoded `href` props below

Reply:
`headerRight: () => <HeaderRight href="/guests/newGuest" />,` will throw this error


```bash
Do not define components during render. React will see a new component type on every render and destroy the entire subtree’s DOM nodes and state (https://reactjs.org/docs/reconciliation.html#elements-of-different-types). Instead, move this component definition out of the parent component “TabLayout” and pass data as props. If you want to allow component creation in props, set allowAsProps option to true.eslint[react/no-unstable-nested-components](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/no-unstable-nested-components.md)
```

Reply:
Define multiple function components outside of the function, and then: 

```ts
headerRight: HeaderRightGuests
```

Etc

Reply:
04aa535b1e0c80ffeb01aa22b8fc88827edb36aa

Reply:
This doesn't look right:

```js
headerRight: () => HeaderRightGuests(),
```

My code above:

```js
headerRight: HeaderRightGuests
```

Reply:
1b324cd6c1393b0883ab59a2de5f2df483974da7